sycl/doc/design/SYCLBINDesign.md

```diff
@@ -0,0 +1,272 @@
+# SYCLBIN - A format for separately compiled SYCL device code
+
+This design document details the SYCLBIN binary format used for storing SYCL
+device binaries to be loaded dynamically by the SYCL runtime. It also details
+how the toolchain produces, links and packages these binaries, as well as how
+the SYCL runtime library handles files of this format.
+
+
+## SYCLBIN binary format
+
+The files produced by the new compilation path will follow the format described
+in this section. The intention of defining a new format for these is to give
+the DPC++ implementation an extendable and lightweight wrapper around the
+multiple modules and corresponding metadata captured in the SYCLBIN file.
+The content of the SYCLBIN may be contained as an entry in the offloading binary
+format produced by the clang-offload-packager, as described in
+[ClangOffloadPackager.rst](https://github.com/intel/llvm/blob/sycl/clang/docs/ClangOffloadPackager.rst).
+
+The format of these files consist of a [file header](#file-header), containing
+general information about the file, followed by three lists of headers: The
+[abstract module header](#abstract-module-header) list, the
+[IR module header](#ir-module-header) list and
+[native device code image header](#native-device-code-image-header) list,
+containing information about the [abstract modules](#abstract-module),
+[IR modules](#ir-module) and
+[native device code images](#native-device-code-image) respectively.
+
+|                                                                       |
+| --------------------------------------------------------------------- |
+| [File header](#file-header)                                           |
```

Comment:
Up until we hit the metadata byte table, everything here is fixed size.  Could we have  "Location|Size" columns added to this list so we can see that file header is 52 bytes, long, the abstract module header is 32 bytes, etc?

That'll help communicate where the metadata byte table should begin, for version 1.